ArticleTemplates/assets/js/bootstraps/campaign.js

```diff
@@ -0,0 +1,35 @@
+define([], function () {
+    'use strict';
+
+    function init() {
+      const campaign = document.querySelector('.campaign--snippet form');
+      if (campaign) {
+        initCampaign(campaign);
+      }
+    }
+
+    function initCampaign(campaign) {
+      campaign.onSubmit(function () {
+        const data = new FormData(campaign);
+        // TODO form submission URL
+        const request = new XMLHttpRequest('POST', '#');
```

Comment:
@alfavata so you're aware, this is the place where I'll need to know where to send the data back to the client app and in which format (currently in `multipart/form-data`)

Reply:
e.g. `new XMLHttpRequest('POST', 'x-gu://callout')`

Reply:
@aodhol can explain how that won't work 😅

Reply:
@regiskuckaertz XHR won't work because Web / native communication doesn't function as client / server rather the native layer intercepts _location_ changes and parses the URL. I'd suggest using JavaScript to gather the data from the various fields constructing an object that is then JSON stringified, URL encoded appended to the URL for us to then parse in the native layer..

Reply:
Yep!

Reply:
@regiskuckaertz once the native layer have exposed an `x-gu://callout` interface for you to pass data to you can use `signalDevice` exposed by `util.js`, this is used elsewhere in the code for passing data to the native layer. 

@aodhol would it be possible for the native layer to expose a JS function on the window now rather than using the `x-gu://callout` method?

Reply:
Whatever works 🤗  as long as it works on both platforms and the app layer catches the call without reloading the page, I'm all good. The only important thing is that I can register callbacks for success and error cases.